### ISSUE TITLE
Validator: check for OpRuntimeArray use being valid

### ISSUE THREAD

**Issue by baldurk** [KhronosGroup/SPIRV-Tools#1871]
I encountered a shader in the wild that had declared `gl_PerVertex.gl_ClipDistance` and `gl_PerVertex.gl_CullDistance` as `OpTypeRuntimeArray` which is doubly invalid - two runtime arrays in the same struct, and runtime arrays on `Output` variables.

Disassembly here:

[invalid.spvas](https://github.com/KhronosGroup/SPIRV-Tools/files/2336044/invalid.txt)

Relevant snippet:

```
%_runtimearr_float = OpTypeRuntimeArray %float
%_runtimearr_float_0 = OpTypeRuntimeArray %float
%gl_PerVertex = OpTypeStruct %v4float %float %_runtimearr_float %_runtimearr_float_0
%_ptr_Output_gl_PerVertex = OpTypePointer Output %gl_PerVertex
          %_ = OpVariable %_ptr_Output_gl_PerVertex Output
```

I ran spirv-val from the 1.1.82.1 SDK and it did sort of complain:

> `error: line 520: According to the Vulkan spec BuiltIn ClipDistance variable needs to be a 32-bit float array. Member #2 of struct ID <332> is not an array.`

but only by coincidence I think - if I remove the BuiltIn decorations it doesn't find any problems.

**Comment by s-perron** [comment]
I'll take a look.  I was looking into adding more validation for OpTypeRuntimeArray as well.

**Comment by s-perron** [comment]
@alan-baker FYI, I did not get to this.  You can get it assigned to somebody else if you need to.

**Comment by baldurk** [comment]
This also needs a check for Uniform non-BufferBlock variables, as in this shader:

```
               OpCapability Shader
               OpMemoryModel Logical Simple
               OpEntryPoint Vertex %main "main"
               OpName %main "main"
               OpDecorate %_runtimearr_v3float ArrayStride 16
               OpDecorate %_struct_3 Block
               OpMemberDecorate %_struct_3 0 Offset 0
               OpDecorate %4 Binding 0
               OpDecorate %4 DescriptorSet 0
       %void = OpTypeVoid
      %float = OpTypeFloat 32
    %v3float = OpTypeVector %float 3
%_runtimearr_v3float = OpTypeRuntimeArray %v3float
  %_struct_3 = OpTypeStruct %_runtimearr_v3float
          %8 = OpTypeFunction %void
%_ptr_Uniform__struct_3 = OpTypePointer Uniform %_struct_3
          %4 = OpVariable %_ptr_Uniform__struct_3 Uniform
       %main = OpFunction %void None %8
         %10 = OpLabel
               OpReturn
               OpFunctionEnd
```

validates clean under `spirv-val repro.spv --target-env vulkan1.1`. With `SPIRV-Tools v2018.7-dev v2018.6-15-ga29a9947` from the 1.1.92.1 SDK.

**Comment by zoddicus** [comment]
These were fixed by #2191.
Both of these examples now generate RTA specific error messages @ ToT.
